### PR TITLE
Queen-Back-Office initialization

### PR DESCRIPTION
**Pull request by kc-bclaudel**

### CODE REVIEW

File: src/main/java/fr/insee/queen/api/configuration/SecurityConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	 * This method configure the HTTP security access
+	 */
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+		http.csrf().disable().headers().frameOptions().disable().and().requestCache()
+				.requestCache(new NullRequestCache());
+		if (isDevelopment()) {
+			switch (this.applicationProperties.getMode()) {
+			case Basic:
+				System.out.println(ApplicationProperties.Mode.Basic);
```
Comment (nicolasTurban) [inline src/main/java/fr/insee/queen/api/configuration/SecurityConfiguration.java]: Remove sysout

File: src/main/java/fr/insee/queen/api/configuration/SwaggerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            }
+            @Override
+            protected String applicationPath() {
+                return "";
+            }
+        };
+    }
+    
+    private ApiInfo apiInfo() {
+        return new ApiInfo("Queen-Back-Office", "Back-office services for Queen", "1.0", "", new Contact("Metallica", "", "dr59-sndi-metallica@outlook.fr"), "LICENSEE", "https://github.com/InseeFr/Queen-Back-Office/blob/master/LICENSE", List.of());
```
Comment (nicolasTurban) [inline src/main/java/fr/insee/queen/api/configuration/SwaggerConfiguration.java]: Remove mail contact for Metallica, prefer link to gihtub issues.

File: src/main/java/fr/insee/queen/api/domain/Comment.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	@Id
+	@GeneratedValue
+	private Long id;
+	
+	/**
+	* The value of comment (jsonb format)
+	*/
+	@Type(type = "jsonb")
+    @Column(columnDefinition = "jsonb")
+	private String value;
```
Comment (nicolasTurban) [inline src/main/java/fr/insee/queen/api/domain/Comment.java]: Should be of type json

File: src/main/resources/application-dev.properties [hunk 4]
```diff
@@ -0,0 +1,24 @@
+environment=dev
+
+server.address=0.0.0.0
+server.port=8080
+server.servlet.context-path=/${spring.application.name}
+
+
+#----base CEI DEV----
```
Comment (nicolasTurban) [inline src/main/resources/application-dev.properties]: Remove mention to CEI.

File: src/main/resources/db/changelog/000_init_test.xml [hunk 5]
```diff
@@ -0,0 +1,110 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog
+	xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+	xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+	<changeSet context="test" author="admin (generated)" id="1584711839954-1">
+        <createSequence sequenceName="hibernate_sequence"/>
+    </changeSet>
+    <changeSet context="test" author="admin (generated)" id="1584714533258-1">
```
Comment (nicolasTurban) [inline src/main/resources/db/changelog/000_init_test.xml]: If it's a test resource it should be put in test/resources ?

File: src/main/resources/db/dev/insert_dev_data.xml [hunk 6]
```diff
@@ -0,0 +1,97 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog
+	xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+	xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+	
+	<!-- Données de test réels -->
```
Comment (nicolasTurban) [inline src/main/resources/db/dev/insert_dev_data.xml]: English.

File: src/main/resources/db/dev/insert_dev_data.xml [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+	</changeSet>
+	
+	<changeSet author="scorcaud" id="testData2" context="dev" objectQuotingStrategy="QUOTE_ONLY_RESERVED_WORDS">
+		<sqlFile dbms="postgresql, oracle" encoding="UTF-8"
+			endDelimiter="\nGO" path="../data/regions_2019_nomenclature.sql"
+			relativeToChangelogFile="true" splitStatements="true"
+			stripComments="true" />
+	</changeSet>
+	
+	<!-- Operation n°1 : simpsons 2020 -->
```
Comment (nicolasTurban) [inline src/main/resources/db/dev/insert_dev_data.xml]: English

File: src/main/resources/db/test/insert_test_data.xml [hunk 8]
```diff
@@ -0,0 +1,97 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
```
Comment (nicolasTurban) [inline src/main/resources/db/test/insert_test_data.xml]: Move to test resources ?

File: src/main/resources/db/test/insert_test_data.xml [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+<databaseChangeLog
+	xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+	xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+	
+	<!-- Données de test réels -->
```
Comment (nicolasTurban) [inline src/main/resources/db/test/insert_test_data.xml]: English

File: src/test/java/fr/insee/queen/api/ApiApplicationTests.java [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+	@Test
+	public void testPutCommentByReportingUnit(){
+		login("admin");
+		ReportingUnit rutest = reportingUnitRepository.getOne((long)22);
+		String value =  "{\"EXTERNAL\": {\"LAST_BROADCAST\": \"12/07/1998\"}}";
+		var commentInit = commentRepository.findDtoByReportingUnit_id(rutest.getId());
+		var put = put("/reporting-unit/22/comment", "{\"EXTERNAL\": {\"LAST_BROADCAST\": \"12/07/1998\"}}", String.class);
+		assert put.getStatusCode().is2xxSuccessful():"Update a comment should be a success";
+		var commentRepo = commentRepository.findDtoByReportingUnit_id(rutest.getId());
+		System.out.println("commentRepo.getvalue : " + commentRepo.getValue());
```
Comment (nicolasTurban) [inline src/test/java/fr/insee/queen/api/ApiApplicationTests.java]: Remove sysout

File: src/test/java/fr/insee/queen/api/ApiApplicationTests.java [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+		login("admin");
+		ReportingUnit rutest = reportingUnitRepository.getOne((long)22);
+		String value =  "{\"EXTERNAL\": {\"LAST_BROADCAST\": \"12/07/1998\"}}";
+		var dataInit = dataRepository.findDtoByReportingUnit_id(rutest.getId());
+		var put = put("/reporting-unit/22/data", "{\"EXTERNAL\": {\"LAST_BROADCAST\": \"12/07/1998\"}}", String.class);
+		assert put.getStatusCode().is2xxSuccessful():"Update a comment should be a success";
+		var dataRepo = dataRepository.findDtoByReportingUnit_id(rutest.getId());
+		System.out.println("dataRepo.getValue() : " +  dataRepo.getValue());
+		System.out.println("dataRepo.getVersion() : " +  dataRepo.getVersion());
+		System.out.println("value : " + value);
```
Comment (nicolasTurban) [inline src/test/java/fr/insee/queen/api/ApiApplicationTests.java]: Sysout to remove

File: src/test/java/fr/insee/queen/api/ApiApplicationTests.java [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+	public void testPutCommentByReportingUnit(){
+		login("admin");
+		ReportingUnit rutest = reportingUnitRepository.getOne((long)22);
+		String value =  "{\"EXTERNAL\": {\"LAST_BROADCAST\": \"12/07/1998\"}}";
+		var commentInit = commentRepository.findDtoByReportingUnit_id(rutest.getId());
+		var put = put("/reporting-unit/22/comment", "{\"EXTERNAL\": {\"LAST_BROADCAST\": \"12/07/1998\"}}", String.class);
+		assert put.getStatusCode().is2xxSuccessful():"Update a comment should be a success";
+		var commentRepo = commentRepository.findDtoByReportingUnit_id(rutest.getId());
+		System.out.println("commentRepo.getvalue : " + commentRepo.getValue());
+		System.out.println("value : " + value);
```
Comment (nicolasTurban) [inline src/test/java/fr/insee/queen/api/ApiApplicationTests.java]: Remove sysout

File: src/test/java/fr/insee/queen/api/ApiApplicationTests.java [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+		assert !commentRepo.getValue().equals(commentInit.getValue()): "Value should should be updated";
```
Comment (nicolasTurban) [inline src/test/java/fr/insee/queen/api/ApiApplicationTests.java]: Doubled should

File: src/test/java/fr/insee/queen/api/ApiApplicationTests.java [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+		ReportingUnit rutest = reportingUnitRepository.getOne((long)22);
+		String value =  "{\"EXTERNAL\": {\"LAST_BROADCAST\": \"12/07/1998\"}}";
+		var dataInit = dataRepository.findDtoByReportingUnit_id(rutest.getId());
+		var put = put("/reporting-unit/22/data", "{\"EXTERNAL\": {\"LAST_BROADCAST\": \"12/07/1998\"}}", String.class);
+		assert put.getStatusCode().is2xxSuccessful():"Update a comment should be a success";
+		var dataRepo = dataRepository.findDtoByReportingUnit_id(rutest.getId());
+		System.out.println("dataRepo.getValue() : " +  dataRepo.getValue());
+		System.out.println("dataRepo.getVersion() : " +  dataRepo.getVersion());
+		System.out.println("value : " + value);
+		assert !dataRepo.getValue().equals(dataInit.getValue()): "Value should should be updated";
```
Comment (nicolasTurban) [inline src/test/java/fr/insee/queen/api/ApiApplicationTests.java]: Should should

File: src/main/java/fr/insee/queen/api/configuration/SecurityConfiguration.java [hunk 15]
```diff
@@ -0,0 +1,149 @@
...
+					.antMatchers("/reporting-unit/{id}/data").hasRole("enqueteur")
+					.antMatchers("/reporting-unit/{id}/comment").hasRole("enqueteur")
+					.antMatchers("/nomenclature/{id}").hasRole("enqueteur")
+					.anyRequest().denyAll(); 
+				break;
+			case Keycloak:
+				System.out.println(ApplicationProperties.Mode.Keycloak);
+				break;
+			default:
+				System.out.println(ApplicationProperties.Mode.NoAuth);
```
Comment (nicolasTurban) [inline src/main/java/fr/insee/queen/api/configuration/SecurityConfiguration.java]: Remove sysou

File: src/main/resources/application-dev.properties (hunk 4 above)
Comment (kc-bclaudel) [reply src/main/resources/application-dev.properties]: corrected

File: src/main/java/fr/insee/queen/api/configuration/SecurityConfiguration.java (hunk 1 above)
Comment (kc-bclaudel) [reply src/main/java/fr/insee/queen/api/configuration/SecurityConfiguration.java]: corrected

File: src/main/java/fr/insee/queen/api/configuration/SwaggerConfiguration.java (hunk 2 above)
Comment (kc-bclaudel) [reply src/main/java/fr/insee/queen/api/configuration/SwaggerConfiguration.java]: fixed

File: src/main/java/fr/insee/queen/api/domain/Comment.java (hunk 3 above)
Comment (kc-bclaudel) [reply src/main/java/fr/insee/queen/api/domain/Comment.java]: fixed

File: src/main/resources/db/dev/insert_dev_data.xml (hunk 6 above)
Comment (kc-bclaudel) [reply src/main/resources/db/dev/insert_dev_data.xml]: fixed

File: src/main/resources/db/test/insert_test_data.xml (hunk 9 above)
Comment (kc-bclaudel) [reply src/main/resources/db/test/insert_test_data.xml]: fixed

File: src/test/java/fr/insee/queen/api/ApiApplicationTests.java (hunk 11 above)
Comment (kc-bclaudel) [reply src/test/java/fr/insee/queen/api/ApiApplicationTests.java]: fixed

File: src/test/java/fr/insee/queen/api/ApiApplicationTests.java (hunk 12 above)
Comment (kc-bclaudel) [reply src/test/java/fr/insee/queen/api/ApiApplicationTests.java]: fixed

File: src/test/java/fr/insee/queen/api/ApiApplicationTests.java (hunk 13 above)
Comment (kc-bclaudel) [reply src/test/java/fr/insee/queen/api/ApiApplicationTests.java]: fixed

File: src/test/java/fr/insee/queen/api/ApiApplicationTests.java (hunk 14 above)
Comment (kc-bclaudel) [reply src/test/java/fr/insee/queen/api/ApiApplicationTests.java]: fixed

File: src/main/java/fr/insee/queen/api/configuration/SecurityConfiguration.java (hunk 15 above)
Comment (kc-bclaudel) [reply src/main/java/fr/insee/queen/api/configuration/SecurityConfiguration.java]: fixed

File: src/main/resources/db/test/insert_test_data.xml (hunk 8 above)
Comment (kc-bclaudel) [reply src/main/resources/db/test/insert_test_data.xml]: this file has been removed

File: src/main/resources/db/changelog/000_init_test.xml (hunk 5 above)
Comment (kc-bclaudel) [reply src/main/resources/db/changelog/000_init_test.xml]: This file has been removed. Usless with the new embedded PosgresSQL database

**Pull request merged**